### PR TITLE
fix(EMS-991): Your business - Turnover - Percentage Turnover allowed commas

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Issue:
* percentageTurnover field allowed commas as valid input

# Changes made:
* Added new helper to handle all validation for percentage numbers
* Removed commas being valid for percentages
* Added tests

### CODE REVIEW

File: src/ui/server/controllers/insurance/business/turnover/validation/rules/percentage-of-turnover.test.ts [hunk 1]
```diff
@@ -32,80 +39,19 @@ describe('controllers/insurance/business/turnover/validation/rules/percentage-of
     });
   });
 
-  describe(`when the ${PERCENTAGE_TURNOVER} input contains a letter`, () => {
-    it('should return a validation error', () => {
-      mockBody[PERCENTAGE_TURNOVER] = 'I5';
+  describe(`when the ${PERCENTAGE_TURNOVER} input is invaluid`, () => {
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/turnover/validation/rules/percentage-of-turnover.test.ts:42]: typo

File: src/ui/server/controllers/insurance/business/turnover/validation/rules/percentage-of-turnover.ts [hunk 2]
```diff
@@ -15,55 +13,28 @@ const {
   EXPORTER_BUSINESS: { [FIELD_ID]: ERROR_MESSAGE },
 } = ERROR_MESSAGES.INSURANCE;
 
-const MINIMUM = 0;
-const MAXIMUM = 100;
-
 /**
  * validates percentage turnover
- * only numbers without decimals or special characters
+ * only numbers without decimals or special characters or commas
```
Comment (ttbarnes) [inline src/ui/server/controllers/insurance/business/turnover/validation/rules/percentage-of-turnover.ts:18]: Very minor but could we reduce the "ors" here 

```js
only numbers without decimals, special characters or commas
```

File: src/ui/types/errors.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+interface Errors {
+  INCORRECT_FORMAT: string;
+  BELOW_MINIMUM: string;
+  ABOVE_MAXIMUM: string;
+}
```
Comment (ttbarnes) [inline src/ui/types/errors.ts:5]: Can we call this `NumberErrorMessage`? More explicit and applies to both non-percentage and percentage type numbers.

File: src/ui/server/helpers/percentage-number-validation/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  const FIELD = 'testField';
+
+  const errorMessages = {
+    INCORRECT_FORMAT: 'Incorrect format',
+    BELOW_MINIMUM: 'Below minimum',
+    ABOVE_MAXIMUM: 'Above maximum',
+  } as Errors;
+
+  describe('percentage is a letter', () => {
+    it('should return a validation error', () => {
```
Comment (abhi-markan) [inline src/ui/server/helpers/percentage-number-validation/index.test.ts:24]: Do we need test cases like percentage is `null`, `undefined` or `empty`?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Good works, one minor change pleas.e

File: src/ui/server/helpers/percentage-number-validation/index.test.ts (hunk 4 above)
Comment (Zainzzkk) [reply src/ui/server/helpers/percentage-number-validation/index.test.ts:24]: no, as that is done in the main validation file, this exclusively checks the inputted value if it exists

Comment (ttbarnes) [reply src/ui/server/helpers/percentage-number-validation/index.test.ts:24]: Good point @Zainzzkk seems we should add `objectHasProperty` validation helper function to the actual validation? Then in the test we can just add "if field is empty" because the helper function handles the rest.

Comment (Zainzzkk) [reply src/ui/server/helpers/percentage-number-validation/index.test.ts:24]: it would mean that this could not be reused on any future optional percentage fields, is that fine?

**Review by ttbarnes — APPROVED** [review]

**Review by abhi-markan — APPROVED** [review]
